lib/facility_factory.rb

```diff
@@ -0,0 +1,69 @@
+class FacilityFactory
+  attr_reader :operating_facilities
+
+  def initialize
+    @operating_facilities = []
+  end
+
+  def define_facilities(facilities)
+    facilities.each do |facility_data|
+      if facility_data[:location_1].is_a?(Hash)
+        address_data = JSON.parse(facility_data[:location_1][:human_address], symbolize_names: true)
+        facility = Facility.new({
+          name: capitalize_words(facility_data[:title]),
+          address: capitalize_words("#{address_data[:address]} #{address_data[:city]} #{address_data[:state]} #{address_data[:zip]}"),
+          phone: facility_data[:phone_number]
+        })
+        @operating_facilities << facility
+      elsif facility_data[:state] == "NY"
+        facility = Facility.new({
+          name: capitalize_words(facility_data[:office_name]),
+          address: capitalize_words("#{facility_data[:street_address_line_1]} #{facility_data[:city]} #{facility_data[:state]} #{facility_data[:zip_code]}"),
+          phone: format_phone_number(facility_data[:public_phone_number]),
+          operating_hours: "Monday - #{facility_data[:monday_beginning_hours]} to #{facility_data[:monday_ending_hours]}, Tuesday - #{facility_data[:tuesday_beginning_hours]} to #{facility_data[:tuesday_ending_hours]}, Wednesday - #{facility_data[:wednesday_beginning_hours]} to #{facility_data[:wednesday_ending_hours]}, Thursday - #{facility_data[:thursday_beginning_hours]} to #{facility_data[:thursday_ending_hours]}, Friday - #{facility_data[:friday_beginning_hours]} to #{facility_data[:friday_ending_hours]}, Sat & Sun - Closed"
+        })
+        @operating_facilities << facility
+      elsif facility_data[:state] == "MO"
+        facility = Facility.new({
+          name: capitalize_words(facility_data[:name]),
+          address: capitalize_words("#{facility_data[:address1]} #{facility_data[:city]} #{facility_data[:state]} #{facility_data[:zipcode]}"),
+          phone: format_phone_number(facility_data[:phone]),
+          operating_hours: capitalize_operating_hours(facility_data[:daysopen]),
+          holiday_closures: facility_data[:holidaysclosed]
+        })
+        @operating_facilities << facility
+      else
+        p "Invalid input, please use list of facilities from relevant states as argument."
+      end
+    end
+    @operating_facilities
+  end
+
+  def capitalize_words(string)
+    excluded_words = ["DMV", "OR", "NY", "MO", "SE", "NE", "SW", "NW"]
+    words = string.split.map do |word|
+      if excluded_words.include?(word)
+        word
+      else
+        word.capitalize
+      end
+    end
+    words.join(" ")
+  end
+
+  def capitalize_operating_hours(hours)
```

Comment:
I don't see tests in the facility_factory_spec.rb file for all these helper methods. Remember, anytime you add a method to a class, we want a test. Yes, these are being tested indirectly when you call your `define_facilites` method, and you DO want to test that, which is called an integration test - making sure everything works together. But then we also want a Unit test for every single method tested in isolation to ensure when we put in a specific input, we get an expected output.